src/net/addr.rs

```diff
@@ -0,0 +1,159 @@
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use std::net::{SocketAddr, SocketAddrV4, SocketAddrV6};
+use std::pin::Pin;
+
+use cfg_if::cfg_if;
+use futures::future::{ready, Ready};
+
+use crate::future::Future;
+use crate::io;
+use crate::task::blocking;
+use crate::task::{Context, Poll};
+
+cfg_if! {
+    if #[cfg(feature = "docs")] {
+        #[doc(hidden)]
+        pub struct ImplFuture<T>(std::marker::PhantomData<T>);
```

Comment:
@skade @stjepang do we really need to play with feature=docs?

Reply:
It makes generated docs look prettier and describes the API's intention better.

We just really want to make docs be as approachable as possible, especially given the fact that docs is an area async programming in Rust has historically not excelled at.